packages/core-web/src/VueCommonAppFactory.js

```diff
@@ -0,0 +1,50 @@
+// eslint-disable-next-line import/no-extraneous-dependencies
+import MarkBindVue from '@markbind/vue-components/src';
+
+/*
+  VueCommonAppFactory provides common data to both the Vue App on server-side and client-side.
+  This is to ensure that our Vue App is a universal application.
+*/
+
+/*
+ These getters are used by popovers and tooltips to get their popover/tooltip content/title.
+
+ For triggers, refer to Trigger.vue.
+ We need to create a completely new popover/tooltip for each trigger due to bootstrap-vue's implementation,
+ so this is how we retrieve our contents.
+*/
+function makeMbSlotGetter(slotName) {
+  return (element) => {
+    const innerElement = element.querySelector(`[data-mb-slot-name="${slotName}"]`);
+    return innerElement === null ? '' : innerElement.innerHTML;
+  };
+}
+
+// Used via vb-popover.html="popoverInnerGetters" for popovers
+const popoverInnerGetters = {
+  title: makeMbSlotGetter('header'),
+  content: makeMbSlotGetter('content'),
+};
+
+// Used via vb-tooltip.html="popoverInnerGenerator" for tooltips
+const tooltipInnerContentGetter = makeMbSlotGetter('_content');
+
+const appFactory = {
```

Comment:
Thanks for the catch! I think this wouldn't cause pollution per se since they are only mainly used for client-side. On the server-side, they are only passed into the Vue App instantiation for SSR so that Vue doesn't warn about undefined (client-side) variables during SSR.